solver/src/main/java/org/chocosolver/solver/propagation/PropagationEngine.java

```diff
@@ -189,6 +189,9 @@ public void propagate() throws ContradictionException {
             manageModifications();
             for (int i = nextNotEmpty(); i > -1; i = nextNotEmpty()) {
                 assert !pro_queue[i].isEmpty() : "try to pop a propagator from an empty queue";
+                if (model.getSolver().isTimeLimitMet()) {
+                    return;
```

Comment:
Stopping the propagation there might lead to thinking the current state is valid / we found a solution. 
Therefore, I would rather throw an exception to make sure the current state is invalid, with something like this 

public final static ICause PROPAG_TIMEOUT = new ICause() {};
model.getSolver().throwsException(PROPAG_TIMEOUT, null, message);

Even if we haven't proved the state is infeasible either...